jung-visualization/src/main/java/edu/uci/ics/jung/visualization/layout/SpatialGridLayoutModel.java

```diff
@@ -14,34 +14,33 @@
 import org.slf4j.LoggerFactory;
 
 /**
- * A LayoutModel that includes a Spatial data structure. This class uses java.awt.geom.Point2D and
- * java2d in order to intersect shapes and determine the content of the spatial grid.
+ * A LayoutModel that includes a SpatialGrid data structure. This class uses java.awt.geom.Point2D
+ * and java2d in order to intersect shapes and determine the content of the spatial grid.
  *
  * @param <N>
  */
-public class SpatialLayoutModel<N> extends LoadingCacheLayoutModel<N, Point2D>
+public class SpatialGridLayoutModel<N> extends LoadingCacheLayoutModel<N, Point2D>
     implements LayoutModel<N, Point2D>, Caching {
 
-  Logger log = LoggerFactory.getLogger(SpatialLayoutModel.class);
+  private static final Logger log = LoggerFactory.getLogger(SpatialGridLayoutModel.class);
 
   protected Spatial<N> spatial;
 
-  public SpatialLayoutModel(
+  public SpatialGridLayoutModel(
       Graph<N> graph, int width, int height, Function<N, Point2D> initializer) {
     super(graph, new AWTPointModel(), width, height, 0, initializer);
```

Comment:
Some thoughts about PointModel and its implementations: 

(0) You don't need the AbstractPointModel class; you can just use Java 8 default methods.  Those defaults could include the offset() methods.
(1) As far as I can tell, AWTPointModel can be a singleton; it's effectively just a stateless bag of methods.  That can save you some allocations.

(It's too bad that we can't just do PointModel.AWT as a static instance, but I realize there's a dependency issue there.)

Reply:
(0) i agree about the existing AbstractPointModel methods. I'm not so sure about providing a tempting easy path for users to skip over the standard library methods of doing things, which should be easier for the JVM to recognize and optimize. Is that just superstition on my part?
(1) It is a singleton in all the samples (I think). I think that a static singleton instance somewhere in jung-visualization would be really useful. I'll make that.